### PR TITLE
Convergence of Antenna control pin names between TX and RX

### PR DESCRIPTION
**Pull request by David-OConnor**
edit: This PR evolved; see comments for details. The rest of this message is the original request; supplanted since this functionality already exists on Tx; PR now makes antenna-select pin names for for Tx and Rx the same, and adds the functionality for Rx.

----

I've noticed that a number of 2-position RF switches use a 2-pin truth table, where 2 input pins must have opposite values. This PR would allow you to connect a second GPIO to the second switcher pin. 

Eg: 
image

Is there anything else in the codebase that needs to be changed for this? Thank you.

An alternative would be to require hardware designers to handle this using *not* gates. Given ELRS boards (or circuits on FCs) tend to be space constrained, I think this would help, at cost of a slight increase to code complexity.

### CODE REVIEW

**Review by pkendall64 — COMMENTED** [review]
To complete this functionality you need to add this pin definition to hardware.h, hardware.cpp, hardware.html, and Unified*.h so it can be used from the unified target definitions.

**Comment by CapnBry** [comment]
I will also say that `_2` is the nomenclature we use for accessing the second set of SX1xxx chips in true diversity mode, so a better name for the variable would be to describe its function instead of just tacking on a 2. `GPIO_PIN_ANTENNA_SELECT_COMPL` for "the complement of antenna select" (INV would be better but I'm concerned that's too close to the "INVERTED" variables we have for inverting the polarity of the primary signal). 

You're also missing setting the output mode of the pin in both rx and tx, and our code convention is to have the opening bracket on the line following the if statement.

**Comment by David-OConnor** [comment]
Thank you. Working on those changes. What is the relation between `RX_ENABLE`/`TX_ENABLE` and the antenna selectors? Can either be used for diversity?

**Comment by David-OConnor** [comment]
Thank you for the detailed feedback. I've attempted to implement both of those sets of changes.

What files should be altered for Tx? I looked through `src/src/tx_main.cpp`, and found `ANT_CTRL` pins. Are these relevant here? Thank you.

**Comment by pkendall64** [comment]
> Actually, I've dug into the code and found what I was looking for. Sorry to be giving you the runaround on this.
> 
> There are GPIO_PIN_ANT_CTRL_1 and GPIO_PIN_ANT_CTRL_2 which currently do this exact function in the TX. I would say we should throw out GPIO_PIN_ANTENNA_SELECT entirely and switch both the TX and RX to use one set of options renamed to GPIO_PIN_ANT_CTRL and GPIO_PIN_ANT_CTRL_COMPL so the functionality is the same between the TX and RX and use the same names.

Yes, this ☝️. We really need to start cleaning house with some of the GPIO pin definitions.

**Comment by David-OConnor** [comment]
Makes sense to me. I think comments on what they do in one or more of those files would help too.

Would you like me to modify this PR accordingly, or handle elsewhere?

Why are there 2 antenna control pins on Tx? Same reason as this?

**Comment by pkendall64** [comment]
> Makes sense to me. I think comments on what they do in one or more of those files would help too.
> 
> Would you like me to modify this PR accordingly, or handle elsewhere?
> 
> Why are there 2 antenna control pins on Tx? Same reason as this?

There is another PR #1662 which adds help text in hardware.html, I think thats the right place for it.

**Comment by David-OConnor** [comment]
Removed `GPIO_PIN_ANTENNA_SELECT` and its complementary; changed logic for both tx and rx to use `GPIO_PIN_ANT_CTRL` and `GPIO_PIN_ANT_CTRL_COMPL`.

After skimming its use in `tx_main`, it appears that indeed, there are 2 pins here for a similar or identical reason I'm requesting in RX, with `_2` being an inversion of `_1`.

**Comment by pkendall64** [comment]
Perhaps rename this PR now to "Convergence of Antenna control pin names between TX and RX".

**Comment by CapnBry** [comment]
I think this is lookin good, but now you need to go through all the target jsons and target includes to be updated with the new names. Examples:
* FM30_RX_MINI.h
  * GPIO_PIN_ANTENNA_SELECT -> GPIO_PIN_ANT_CTRL
  * GPIO_PIN_ANT_CTRL_1 -> GPIO_PIN_ANT_CTRL
  * Since these are now the same they can be pulled out of the #ifdef so there's just one define, progress!
* Generic 2400 PA Diversity.json
  * ant_select -> ant_ctrl

Basically update anywhere it had the old names with the new names.

**Comment by David-OConnor** [comment]
Done (for files in those 2 folders)

**Comment by CapnBry** [comment]
Looks good! @pkendall64 Should we target this for 3.0 instead? Considering it is the first release with the hardware.json, I would consider it to be a good idea to not have to rename the fields in 3.1. We have a lot of hardware coming that are going to be using these pins and you know how infrequently our partners update their git, so we could be dealing with JSONs coming for a year that use `ant_select`.

Seems to me that locking the new names now before the doomed names get a chance to start propagating in the wild would be important for reducing naming confusion later. We still have the chance for things to be named correctly now!

**Comment by pkendall64** [comment]
> Looks good! @pkendall64 Should we target this for 3.0 instead? Considering it is the first release with the hardware.json, I would consider it to be a good idea to not have to rename the fields in 3.1. We have a lot of hardware coming that are going to be using these pins and you know how infrequently our partners update their git, so we could be dealing with JSONs coming for a year that use `ant_select`.
> 
> Seems to me that locking the new names now before the doomed names get a chance to start propagating in the wild would be important for reducing naming confusion later. We still have the chance for things to be named correctly now!

Yes, I wholeheartedly agree! We will have to fix the conflicts if we target it for 3.0.

**Comment by pkendall64** [comment]
I've rebased this to 3.x and retargeted to 3.0 release as we want to try to avoid too many changes to hardware definitions after 3.0 is released.

**Review by pkendall64 — APPROVED** [review]

**Review by CapnBry — APPROVED** [review]
Thanks, Paul! I'm happy we merged these variables since it has bothered me and been a little unclear what they did for as long as they've been in.

I'd like to do the same with PIN_LED (generally RX only) vs PIN_LED_GREEN vs PIN_LED_RED maybe for 3.1, since as we've unified more code it has become crazy confusing what the function of these items is intended to do. The functionality of each has changed subtly as we refactor and it would be better to have a primary led and an optional secondary led.
